### PR TITLE
WEA-6 Home Screen Layout Implementation

### PR DESCRIPTION
**Pull request by ayush-shta**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,7 +10,10 @@
       android:roundIcon="@mipmap/ic_launcher_round"
       android:supportsRtl="true"
       android:theme="@style/AppTheme">
-    <activity android:name="com.ayush.weatherapp.splash.MainActivity">
+    <activity
```
Comment (subash-poudel) [inline app/src/main/AndroidManifest.xml]: should we have orientation portrait here?

File: app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    TypedArray typedArray =
+        getContext().obtainStyledAttributes(attributeSet, R.styleable.ForecastCompoundView);
+
+    setValues(typedArray);
+
+    typedArray.recycle();
+  }
+
+  private void setValues(TypedArray typedArray) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java:53]: can we use the setter like setTopText() and provide the value fetched from typed array.

File: app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    tvTop.setText(typedArray.getString(R.styleable.ForecastCompoundView_day));
+    ivMid.setImageDrawable(
+        typedArray.getDrawable(R.styleable.ForecastCompoundView_weather_icon));
+    tvBottom.setText(typedArray.getString(R.styleable.ForecastCompoundView_temperature));
+  }
+
+  public TextView getTvTop() {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java:60]: no need to return the view. this is redundant

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  @BindView(R.id.layout_home) RelativeLayout layoutHome;
+  @BindView(R.id.grp_list_forecast) LinearLayout grpListForecast;
+
+  @BindView(R.id.detail_sun) ForecastDetailCompoundView detailSun;
+  @BindView(R.id.detail_wind) ForecastDetailCompoundView detailWind;
+  @BindView(R.id.detail_temperature) ForecastDetailCompoundView detailTemperature;
+
+  HomeContract.Presenter presenter;
+
+  @Override protected int getContextView() {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java:52]: getLayoutId

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    return R.layout.activity_home;
+  }
+
+  @Override protected void onCreate(Bundle savedInstanceState) {
+    //override theme defined in the xml for splash screen effect
+    setTheme(R.style.AppTheme);
+
+    super.onCreate(savedInstanceState);
+
+    layoutHome.setVisibility(View.INVISIBLE);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java:62]: w u do dis?

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+  @Override public void hideProgressDialog() {
+    getProgressDialog().dismiss();
+  }
+
+  @Override public void setCurrentForecast(double latitude, double longitude,
+      CurrentForecast currentForecast) {
+    layoutHome.setVisibility(View.VISIBLE);
+
+    tvLocation.setText(LocationUtils.getLocality(latitude, longitude, this));
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java:120]: why is context required?

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  @Override public HomeContract.View getView() {
+    return view;
+  }
+
+  @Override public void fetchWeatherDetails() {
+    getView().showProgressDialog("Loading", false);
+
+    APIInterface apiInterface = APIClient.getClient().create(APIInterface.class);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java:31]: Repository pattern study

File: app/src/main/java/com/ayush/weatherapp/home/pojo/DailyForecast.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    public double getApparentTemperatureLow() {
+      return apparentTemperatureLow;
+    }
+
+    public double getWindSpeed() {
+      return windSpeed;
+    }
+  }
+}
+
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/pojo/DailyForecast.java:76]: only one enter

File: app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+  public static  String getDayOfTheWeek(int timeStamp) {
+
+    if (isToday((long) timeStamp * 1000)) {
+      return "Today";
+    }
+
+    Date date = new Date((long) timeStamp * 1000);
+
+    return new SimpleDateFormat("EE", Locale.getDefault()).format(date);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java:22]: DAY_OF_WEEK = "EE"

File: app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  public static String getTime(int timeStamp) {
+    Date date = new Date((long) timeStamp * 1000);
+
+    return new SimpleDateFormat("hh:mm aa", Locale.getDefault()).format(date);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java:28]: .

File: app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+import com.ayush.weatherapp.R;
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+public final class LocationUtils {
+
+  public static String getLocality(double latitude, double longitude, Context context) {
+
+    Geocoder geocoder = new Geocoder(context, Locale.getDefault());
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/LocationUtils.java:15]: blasphemy

File: app/src/main/java/com/ayush/weatherapp/utils/MapperUtils.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
+package com.ayush.weatherapp.utils;
+
+import com.ayush.weatherapp.R;
+
+public final class MapperUtils {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/MapperUtils.java:5]: file name inappropriate

File: app/src/main/res/layout/layout_home.xml [hunk 13]
```diff
@@ -0,0 +1,149 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:scrollbars="vertical"
+    >
+
+  <!--Current Forecast group-->
+  <RelativeLayout
```
Comment (subash-poudel) [inline app/src/main/res/layout/layout_home.xml:10]: unnecessary

File: app/src/main/res/layout/layout_home.xml [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+        android:layout_centerHorizontal="true"
+        android:minHeight="@dimen/img_weather_min_height"
+        android:src="@drawable/img_cloudy_day"
+        />
+
+  </RelativeLayout>
+
+
+  <!--Horizontal line-->
+  <View
```
Comment (subash-poudel) [inline app/src/main/res/layout/layout_home.xml:59]: use include

File: app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java (hunk 2 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java:53]: :ok:

File: app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java (hunk 3 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/customViews/ForecastCompoundView.java:60]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java (hunk 4 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java:52]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java (hunk 5 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java:62]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/pojo/DailyForecast.java (hunk 8 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/pojo/DailyForecast.java:76]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java (hunk 9 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java:22]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java (hunk 10 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/DateUtils.java:28]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/MapperUtils.java (hunk 12 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/MapperUtils.java:5]: :ok:

File: app/src/main/res/layout/layout_home.xml (hunk 13 above)
Comment (ayush-shta) [reply app/src/main/res/layout/layout_home.xml:10]: :ok:

File: app/src/main/res/layout/layout_home.xml (hunk 14 above)
Comment (ayush-shta) [reply app/src/main/res/layout/layout_home.xml:59]: :ok:

**Pull request closed without merging**